examples/online-shop/service.bal

```diff
@@ -0,0 +1,149 @@
+        stream<Product, error?> resultStream = dbClient->query(`SELECT * FROM PRODUCTS WHERE PRODUCT_ID = ${id}`);
+        record {|Product value;|}? data = check resultStream.next();
+        check resultStream.close();
+        return data?.value;
+    }
+
+    resource function get customers() returns Customer[]|error? {
+        Customer[] customers = [];
+        stream<Customer, error?> resultStream = dbClient->query(`SELECT * FROM CUSTOMERS`);
+        _ = check resultStream.forEach(function(Customer customer) {
+           customers.push(customer);
+        });
+        check resultStream.close();
+        return customers;
+    }
+
+    resource function get customers/[decimal id]() returns Customer|error? {
+        stream<Customer, error?> resultStream = dbClient->query(`SELECT * FROM CUSTOMERS WHERE CUSTOMER_ID = ${id}`);
+        record {|Customer value;|}? data = check resultStream.next();
+        check resultStream.close();
+        return data?.value;
+    }
+
+    resource function get orders() returns Order[]|error? {
+        Order[] orders = [];
+        stream<Order, error?> resultStream = dbClient->query(`SELECT * FROM ORDERS`);
+        _ = check resultStream.forEach(function(Order order) {
+           orders.push(order);
+        });
+        check resultStream.close();
+        return orders;
+    }
+
+    resource function get orders/[decimal id]() returns Order|error? {
+        stream<Order, error?> resultStream = dbClient->query(`SELECT * FROM ORDERS WHERE ORDER_ID = ${id}`);
+        record {|Order value;|}? data = check resultStream.next();
+        check resultStream.close();
+        return data?.value;
+    }
+
+    resource function get orderItems() returns OrderItem[]|error? {
+        OrderItem[] orderItems = [];
+        stream<OrderItem, error?> resultStream = dbClient->query(`SELECT * FROM ORDER_ITEMS`);
+        _ = check resultStream.forEach(function(OrderItem orderItem) {
+           orderItems.push(orderItem);
+        });
+        check resultStream.close();
+        return orderItems;
+    }
+
+    resource function get orderItemsByOrderId/[decimal id]() returns OrderItem[]|error? {
+        OrderItem[] orderItems = [];
+        stream<OrderItem, error?> resultStream = dbClient->query(`SELECT * FROM ORDER_ITEMS WHERE ORDER_ID = ${id}`);
+        _ = check resultStream.forEach(function(OrderItem orderItem) {
+           orderItems.push(orderItem);
+        });
+        check resultStream.close();
+        return orderItems;
+    }
+
+    resource function get contacts() returns Contact[]|error? {
+        Contact[] contacts = [];
+        stream<Contact, error?> resultStream = dbClient->query(`SELECT * FROM CONTACTS`);
+        _ = check resultStream.forEach(function(Contact contact) {
+           contacts.push(contact);
+        });
+        check resultStream.close();
+        return contacts;
+    }
+
+    resource function get contactsByCustomerId/[decimal id]() returns Contact[]|error? {
```

Comment:
why the `id` type is decimal?

Reply:
change id to int in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/36/commits/9c39326fb639c7553dd84fc179accd374e46f2ff